src/index.ts

```diff
@@ -29,56 +29,62 @@ export * from './models'
+  opts: ApiOptions = {
+    url: '127.0.0.1',
+    port: 40600,
+    version: 0
+  }
   /** @property {Account} account - Manage Textile wallet account */
-  account: Account
+  account: Account = new Account(this.opts)
   /** @property {Blocks} blocks - Manage Textile Blocks */
-  blocks: Blocks
+  blocks: Blocks = new Blocks(this.opts)
   /** @property {Cafes} account - Manage Textile Cafes */
-  cafes: Cafes
+  cafes: Cafes = new Cafes(this.opts)
   /** @property {Config} config - Manage Textile Config settings */
-  config: Config
+  config: Config = new Config(this.opts)
   /** @property {Comments} comments - Manage Textile block Comments */
-  comments: Comments
+  comments: Comments = new Comments(this.opts)
   /** @property {Contacts} contacts - Manage Textile peer Contacts */
-  contacts: Contacts
+  contacts: Contacts = new Contacts(this.opts)
   /** @property {Feed} feed - Manage the Textile Feed */
-  feed: Feed
+  feed: Feed = new Feed(this.opts)
   /** @property {File} file - Manage a Textile File */
-  file: File
+  file: File = new File(this.opts)
   /** @property {Files} files - Manage Textile Files */
-  files: Files
+  files: Files = new Files(this.opts)
   /** @property {Invites} invites - Manage Textile Invites */
-  invites: Invites
+  invites: Invites = new Invites(this.opts)
   /** @property {IPFS} ipfs - Manage the underlying IPFS peer */
-  ipfs: IPFS
+  ipfs: IPFS = new IPFS(this.opts)
   /** @property {Likes} likes - Manage Textile block Likes */
-  likes: Likes
+  likes: Likes = new Likes(this.opts)
   /** @property {Logs} logs - Manage Textile subsystem logs */
-  logs: Logs
+  logs: Logs = new Logs(this.opts)
   /** @property {Messages} messages - Manage Textile thread Messages */
-  messages: Messages
+  messages: Messages = new Messages(this.opts)
   /** @property {Notifications} notifications - Manage Textile Notifications */
-  notifications: Notifications
+  notifications: Notifications = new Notifications(this.opts)
   /** @property {Profile} profile - Manage a Textile node's public profile */
-  profile: Profile
+  profile: Profile = new Profile(this.opts)
   /** @property {Schemas} schemas - Manage Textile Schemas */
-  schemas: Schemas
+  schemas: Schemas = new Schemas(this.opts)
   /** @property {Subscribe} subscribe - Subscribe to thread updates */
-  subscribe: Subscribe
+  subscribe: Subscribe = new Subscribe(this.opts)
   /** @property {Threads} threads - Manage Textile Threads */
-  threads: Threads
+  threads: Threads = new Threads(this.opts)
   /** @property {Tokens} tokens - Manage Textile Threads */
-  tokens: Tokens
+  tokens: Tokens = new Tokens(this.opts)
   /** @property {Utils} utils - Get information about the Textile node */
-  utils: Utils
-  constructor(options?: ApiOptions) {
+  utils: Utils = new Utils(this.opts)
+
+/**
+ * Textile is the main client class
+ *
+ * @param {TextileOptions} options The Textile options object
+ */
+  setOptions(options?: TextileOptions) {
```

Comment:
setOptions takes our new type, TextileOptions where port, url, version are all optionals and then just merges with the defaults